### PR TITLE
Allow overriding the current time

### PR DESCRIPTION
**Pull request by zupa-hu**
For deterministic results, I need to control access to the current time.

There is already a way to control access to the random source via `Runtime.SetRandSource()`. This PR adds `Runtime.SetCurrentTimeGetter()` that works similarly.

### CODE REVIEW

File: runtime.go [hunk 1]
```diff
@@ -1421,6 +1425,12 @@ func (r *Runtime) SetRandSource(source RandSource) {
 	r.rand = source
 }
 
+// SetCurrentTimeGetter sets the current time getter function for this Runtime.
+// If not called, the default time.Now() is used.
+func (r *Runtime) SetCurrentTimeGetter(now Now) {
```
Comment (dop251) [inline runtime.go]: Could you please rename it to SetTimeSource()?

**Comment by zupa-hu** [comment]
Sure, on it.

File: runtime.go (hunk 1 above)
Comment (zupa-hu) [reply runtime.go]: Done, I just have to figure out how to update the PR.

**Comment by zupa-hu** [comment]
Thanks!

**Pull request merged**